.jenkinsci/github-api.groovy

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env groovy
+
+def mergePullRequest() {
+	if ( ! ( checkMergeAcceptance() ) ) { return false  }
+	withCredentials([string(credentialsId: 'jenkins-integration-test', variable: 'sorabot')]) {
+		def slurper = new groovy.json.JsonSlurperClassic()
+		def commitTitle = ""
+		def commitMessage = ""
+		def mergeMethod = getMergeMethod()
+		def jsonResponseMerge = sh(script: """
+		curl -H "Authorization: token ${sorabot}" \
+				 -H "Accept: application/vnd.github.v3+json" \
+				 -X PUT --data '{"commit_title":"${commitTitle}","commit_message":"${commitMessage}","sha":"${env.GIT_COMMIT}","merge_method":"${mergeMethod}"}' \
+				 -w "%{http_code}" https://api.github.com/repos/hyperledger/iroha/pulls/${CHANGE_ID}/merge""", returnStdout: true)
+		def githubResponse = sh(script:"""set +x; printf '%s\n' "${jsonResponseMerge}" | tail -n 1; set -x""", returnStdout: true).trim()
+		jsonResponseMerge = slurper.parseText(jsonResponseMerge)
+		if (jsonResponseMerge.merged != "true" || !(githubResponse ==~ "200")) {
+			echo jsonResponseMerge.message
+			return false
+		}
+		return true
+	}
+}
+
+def checkMergeAcceptance() {
+	def approvalsRequired = 2
+	def gitCommitterEmail = sh(script: 'git --no-pager show -s --format=\'%ae\'', returnStdout: true).trim()
+	wrap([$class: 'BuildUser']) {
+		jenkinsCommitterEmail = env.BUILD_USER_EMAIL
+	}
+  withCredentials([string(credentialsId: 'jenkins-integration-test', variable: 'sorabot')]) {
+  	def slurper = new groovy.json.JsonSlurperClassic()
+  	def jsonResponseReview = sh(script: """
+  		curl -H "Authorization: token ${sorabot}" -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/hyperledger/iroha/pulls/${CHANGE_ID}/reviews
+			""", returnStdout: true).trim()
+		jsonResponseReview = slurper.parseText(jsonResponseReview)
+		if (jsonResponseReview.size() > 0) {
+			jsonResponseReview.each {
+				if ("${it.state}" == "APPROVED") {
```

Comment:
Also, we must forbid merge when someone "request changes" for the pull request.

Reply:
@muratovv good point, I will add this feature at the next commit